Tools/src/FormatCheckTool/FormatCheckTool.cs

```diff
@@ -0,0 +1,234 @@
+            {
+                return;
+            }
+
+            bool stage3 = CheckModelResult(data.Item2);
+        }
+
+        private static Tuple<bool, List<TestModel>> LoadJsonFile(string filePath)
+        {
+            List<TestModel> specs = new List<TestModel>{ };
+            string flag = string.Empty;
+            if (!File.Exists(filePath))
+            {
+                flag = $"File {filePath} Dose Not Exist! {PleaseCheckStr}";
+            }
+            else
+            {
+                var rawData = File.ReadAllText(filePath);
+
+                // Try to load Specs JSON file, and catch exception.
+                try
+                {
+                    specs = (List<TestModel>)JsonConvert.DeserializeObject<IList<TestModel>>(rawData);
+                    if (!specs.Any())
+                    {
+                        flag = $"JSON File is Empty, {PleaseCheckStr}";
+                    }
+                }
+                catch (JsonReaderException jre)
+                {
+                    flag = $"JSON Reader Exception! {PleaseCheckStr}";
+                }
+                catch (JsonSerializationException jse)
+                {
+                    flag = $"JSON SerialzationException! {PleaseCheckStr}";
+                }
+                catch (Exception e)
+                {
+                    flag = $"Other File Error, {PleaseCheckStr}";
+                }
+            }
+
+            return new Tuple<bool, List<TestModel>>(ConsoleStageResult("JSON Valid", flag), specs);
+        }
+
+        private static bool CheckBasicContent(List<TestModel> data)
+        {
+            List<string> res = new List<string> { };
+            foreach (TestModel spec in data)
+            {
+                res.Add(CheckOneBasicContentSpec(spec));
+            }
+
+            return ConsoleStageResult("Content Valid", string.Join("|", res), true);
+        }
+
+        private static bool CheckModelResult(List<TestModel> data)
+        {
+            string flag = string.Empty;
+            return ConsoleStageResult("Model Result", flag);
+        }
+
+        private static string CheckOneBasicContentSpec(TestModel spec)
+        {
+            string flag = string.Empty;
+            if (spec.Input.Equals(string.Empty))
+            {
+                flag = "No Input";
+            }
+            else
+            {
+                foreach (object result in spec.Results)
+                {
+                    string resultStr = CheckOneBasicContentSpecResult(spec, result);
+                    if (!resultStr.Equals(string.Empty))
+                    {
+                        flag = resultStr;
+                        break;
+                    }
+                }
+            }
+
+            return flag;
+        }
+
+        private static string CheckOneBasicContentSpecResult(TestModel spec, object result)
+        {
+            var serializeResult = JsonConvert.SerializeObject(result);
```

Comment:
Why do we need to serialize and deserialize something here? Isn't the spec already deserialized at this stage?